File: widgets/lib_arch/src/main/java/com/yyxnb/lib_arch/annotations/BindViewModel.java
package com.yyxnb.lib_arch.annotations;

import androidx.lifecycle.ViewModel;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * 绑定 {@link ViewModel}
 *
 * @author yyx
 */
@Target(ElementType.FIELD)
@Retention(RetentionPolicy.RUNTIME)
public @interface BindViewModel {
    boolean isActivity() default false;
}

File: lib_arch/src/main/java/com/yyxnb/arch/annotations/BindViewModel.kt
package com.yyxnb.arch.annotations

/**
 * 绑定 [androidx.lifecycle.ViewModel]
 *
 * @author yyx
 */
@Target(AnnotationTarget.FIELD)
@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
annotation class BindViewModel(val isActivity: Boolean = false)